test/test_fx.py

```diff
@@ -563,6 +563,165 @@ def test_construct_root_dict(self):
+                        if isinstance(value, torch.fx.Node):
+                            new_kwargs[key] = new_nodes[value.name]
+                        else:
+                            new_kwargs[key] = value
+
+                    # create target
+                    target = (
+                        module_name + "." + orig_node.name
+                        if orig_node.op in ["get_param", "call_module"]
+                        else orig_node.target
+                    )
+
+                    # create new node
+                    new_nodes[node_name] = graph.create_node(
+                        op=orig_node.op,
+                        target=target,
+                        args=new_args,
+                        kwargs=new_kwargs,
+                        name=orig_node.name,
+                    )
+                    if node_name == module_node_names[-1]:
+                        graph.output(new_nodes[node_name])
+
+                    if orig_node.op == "placeholder":
+                        graph_placeholders.append(new_nodes[node_name])
+
+                # set up dict for copying over targets
+                subgraph_targets = {}
+                for node_name, node in new_nodes.items():
+                    if node.op in ["get_param", "call_module"]:
+                        if node_name in submodule_hints:
+                            # when node is a module created from this function, target is
+                            # in orig_node's target directly
+                            node_target = orig_nodes[node_name].target
+                        else:
+                            # the actual target still lives in the original torch.nn.module's hiearchy
+                            node_target = orig_base_module
+                            for atom in orig_nodes[node_name].target.split("."):
+                                node_target = getattr(node_target, atom)
+                        subgraph_targets[str(node.target)] = node_target
+
+                module: torch.fx.GraphModule = torch.fx.GraphModule(subgraph_targets, graph)
+                module_as_node = torch.fx.Node(
+                    graph=graph,
+                    name=module_name,
+                    op="call_module",
+                    target=module,
+                    args=tuple(graph_placeholders),
+                    kwargs={},
+                )
+
+                return (module, module_as_node)
+
+            # get nodes from original GraphModule
+            orig_nodes: Dict[str, torch.fx.Node] = {}
+            for node in orig_graph_module.graph.nodes:
+                orig_nodes[node.name] = node
+
+            # create subgraph A
+            subgraph_a_partition = ["x", "param", "add_1", "linear_1", "clamp_1"]
+            submodule_a, submodule_node_a = create_module_from_nodes(
+                "submod_a", subgraph_a_partition, orig_nodes, orig_torch_nn_module
```

Comment:
Nit: using `kwargs` at the callsite of a function with many arguments (3 or more in my preference, or 2 if the arguments are easily confused) can make the code easier to understand, e.g.

```
submodule_a, submodule_node_a = create_module_from_nodes(
    module_name="submod_a", module_name_nodes=subgraph_a_partition, orig_nodes=orig_nodes, orig_base_module=orig_torch_nn_module)
```